Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2.py

```diff
@@ -0,0 +1,4180 @@
+                {"XsoarUser": user, "SplunkUser": splunk_user or "Could not map splunk user, Check logs for more info."}
+            )
+
+        return CommandResults(
+            outputs=outputs,
+            outputs_prefix="Splunk.UserMapping",
+            readable_output=tableToMarkdown("Xsoar-Splunk Username Mapping", outputs, headers=["XsoarUser", "SplunkUser"]),
+        )
+
+    def update_xsoar_user_in_findings(self, findings_data):
+        """In case of `should_map_user` is True, update the 'owner' in the findings to be the mapped XSOAR user.
+
+        Args:
+            findings_data (list[dict]): The findings to be updated.
+        """
+        if self.should_map:
+            demisto.debug("UserMapping: instance configured to map Splunk user to XSOAR users, trying to map.")
+            for finding_data in findings_data:
+                if splunk_user := finding_data.get("owner"):
+                    xsoar_user = self.get_xsoar_user_by_splunk(splunk_user)
+                    finding_data["owner"] = xsoar_user
+                    demisto.debug(
+                        f"UserMapping: 'owner' was mapped from {splunk_user} to {xsoar_user} "
+                        f"for finding {finding_data.get(EVENT_ID)}."
+                    )
+
+
+class SplunkGetModifiedRemoteDataResponse(GetModifiedRemoteDataResponse):
+    """get-modified-remote-data response parser
+
+    :type modified_findings_data: ``list``
+    :param modified_findings_data: The Findings that were modified since the last check.
+
+    :type entries: ``list``
+    :param entries: The entries you want to add to the war room.
+
+    :return: No data returned
+    :rtype: ``None``
+    """
+
+    def __init__(self, modified_findings_data, entries):
+        self.modified_findings_data = modified_findings_data
+        self.entries = entries
+        extensive_log(f"mirror-in: updated findings: {self.modified_findings_data}")
+        extensive_log(f"mirror-in: updated entries: {self.entries}")
+
+    def to_entry(self):
+        """Convert data to entries.
+
+        :return: List of findings data as entries + entries (from comments and close data),
+                 or [{}] if there are only entries and no modified findings.
+        :rtype: ``list``
+        """
+        findings_entries = [
+            {
+                "EntryContext": {"mirrorRemoteId": data[RULE_ID]},
+                "Contents": data,
+                "Type": EntryType.NOTE,
+                "ContentsFormat": EntryFormat.JSON,
+            }
+            for data in self.modified_findings_data
+        ]
+
+        if not findings_entries and self.entries:
+            return [{}] + self.entries
+
+        return findings_entries + self.entries
+
+
+# =========== Time & Date Utilities ===========
+
+
+def get_current_splunk_time(splunk_service: client.Service):
+    get_time_query = f'| gentimes start=-1 | eval clock = strftime(time(), "{ISO_FORMAT_TZ_AWARE}") | sort 1 -_time | table clock'
```

Comment:
can you please a coment about what this query is getting?

Reply:
added as part of the doc string